elastic/kibana
# [Security AI Assistant] Update anonymization privileges

Added new sub-feature to AI-Assistant which allows to grant user role privilege to edit Anonymization fields:
<img width="761" alt="Screenshot 2024-04-11 at 7 29 50 PM" src="https://github.com/elastic/kibana/assets/2946766/d4358178-d8e9-4c68-b7d4-a19d2befa29b">



How to test:
1. Create user role, which has access to Security and Actions/Connectors and AI Assistant.
2. Customize sub-feature privileges: Remove checkbox for Update anonymization fields.
3. Save role.
4. Create/update existing user with the new role.
5. Go to Security AI Assistant settings and open Anonymization tab.
6. For the user role with removed privilege to edit anonymization, all actionable buttons should be disabled. Button Save will remain enabled, because it correspond to all Assistant settings.
7. Public API "/api/elastic_assistant/anonymization_fields/_bulk_action" should return "Forbidden" 403 access error.

How it looks when no privilege for the user role:
<img width="771" alt="Screenshot 2024-04-12 at 10 12 34 AM" src="https://github.com/elastic/kibana/assets/55110838/7c3b6c92-12cb-46ae-8356-dc687c82726f">

@patrykkopycinski please adopt it after merging this PR to the new UX
